pkg/virt-controller/services/template.go

```diff
@@ -194,6 +218,52 @@ func (t *templateService) RenderLaunchManifest(vm *v1.VirtualMachine) (*kubev1.P
 	return &pod, nil
 }
 
+// setMemoryOverhead computes the estimation of total
+// memory needed for the domain to operate properly.
+// This includes the memory needed for the guest and memory
+// for Qemu and OS overhead.
+//
+// The return values are requested memory and limit memory quantities
+//
+// Note: This is the best estimation we were able to come up with
+//       and is still not 100% accurate
+func setMemoryOverhead(domain v1.DomainSpec, resources *kubev1.ResourceRequirements) error {
+	vmMemoryReq := domain.Resources.Requests.Memory()
+
+	overhead := resource.NewScaledQuantity(0, resource.Kilo)
+
+	// Add the memory needed for pagetables (one bit for every 512b of RAM size)
+	pagetableMemory := resource.NewScaledQuantity(vmMemoryReq.ScaledValue(resource.Kilo), resource.Kilo)
+	pagetableMemory.Set(pagetableMemory.Value() / 512)
+	overhead.Add(*pagetableMemory)
+
+	// Add fixed overhead for shared libraries and such
+	// TODO account for the overhead of kubevirt components running in the pod
+	overhead.Add(*resource.NewScaledQuantity(64, resource.Mega))
+
+	// Add CPU table overhead (8 MiB per vCPU and 8 MiB per IO thread)
```

Comment:
Right now we only support setting vCPUs. IO threads and sockets is one. @MarSik does that mean that we have to do `8Mi + 8Mi` for one cpu with one trhead?